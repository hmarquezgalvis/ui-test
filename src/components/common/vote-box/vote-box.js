import React, { useState } from 'react';
import styled from 'styled-components';

import { VoteButton } from '../vote/vote';

export const VoteBox = ({ guid, voted = false, updateVote = () => {} }) => {
  const values = {
    thumbUp: 1,
    thumbDown: -1,
  };
  const [selected, setSelected] = useState(null);

  const onVoteSelected = (value) => {
    if (selected === value) {
      setSelected(null);
      return;
    }
    setSelected(value);
  };
  const isActive = (value) => value === selected;
  const voteNowIsDisabled = selected === null;

  return (
    <VoteBoxContainer>
      {voted ? (
        <Button
          onClick={() => { updateVote(guid, null); setSelected(null); }}
        >Vote Again</Button>
      ) : (
        <>
          <VoteButton
            selected={isActive(values.thumbUp)}
            value={values.thumbUp}
            onClick={onVoteSelected} /><VoteButton
            thumbDown
            selected={isActive(values.thumbDown)}
            value={values.thumbDown}
            onClick={onVoteSelected}
          >
            <VoteButton.ThumbDown />
          </VoteButton>
          <Button
            disabled={voteNowIsDisabled}
            onClick={() => updateVote(guid, selected)}
          >Vote Now</Button>
        </>
      )}
    </VoteBoxContainer>
  );
};

const VoteBoxContainer = styled.div`
  display: flex;
  gap: 12px;
  justify-content: flex-end;
  align-items: center;
  margin-bottom: 12px;
  color: #FFFFFF;
`;

const Button = styled.button`
  min-width: 100px;
  height: 38px;
  color: #FFFFFF;
  font-size: 15px;
  line-height: 18px;
  text-align: center;
  vertical-align: middle;
  background: rgba(0, 0, 0, 0.6);
  border: 1px solid #FFFFFF;
  cursor: pointer;

  &:disabled {
    cursor: default;
    color: #999999;
  }
`;