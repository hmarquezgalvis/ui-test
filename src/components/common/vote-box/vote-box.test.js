import { render } from '@testing-library/react';
import { VoteBox } from './vote-box';

describe('Tests for <VoteBox />.', () => {

  describe('should render and save the snapshot correctly.', () => {

    it('default state', () => {
      const { asFragment } = render(
        <VoteBox />
      );
      expect(asFragment()).toMatchSnapshot();
    });

    it('voted state', () => {
      const { asFragment } = render(
        <VoteBox voted />
      );
      expect(asFragment()).toMatchSnapshot();
    });

  });

});