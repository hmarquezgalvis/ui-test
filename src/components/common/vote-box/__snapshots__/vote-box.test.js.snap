// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <VoteBox />. should render and save the snapshot correctly. default state 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf hhMSm"
  >
    <button
      class="sc-bczRLJ hebTOi"
    >
      <img
        alt=""
        src="thumbs-up.svg"
      />
    </button>
    <button
      class="sc-bczRLJ dgfLAZ"
    >
      <img
        alt=""
        src="thumbs-down.svg"
      />
    </button>
    <button
      class="sc-hKMtZM RivLz"
      disabled=""
    >
      Vote Now
    </button>
  </div>
</DocumentFragment>
`;

exports[`Tests for <VoteBox />. should render and save the snapshot correctly. voted state 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf hhMSm"
  >
    <button
      class="sc-hKMtZM RivLz"
    >
      Vote Again
    </button>
  </div>
</DocumentFragment>
`;
